Resources/config/dbal.xml

```diff
@@ -23,19 +20,6 @@
         <service id="Doctrine\DBAL\Driver\Connection" alias="database_connection" public="false" />
         <service id="Doctrine\DBAL\Connection" alias="database_connection" public="false" />
 
-        <service id="doctrine.dbal.logger.chain" class="%doctrine.dbal.logger.chain.class%" public="false" abstract="true" >
```

Comment:
AFAICS this is a BC break isn't it? Although the service has been deprecated (not sure it actually has been marked as such), this was supposedly done in 2.3.0

Reply:
We favored BC break over userspace hooking into code that does nothing, which would cause functionality to silently stop working. BTW we don't guarantee BC promise on DI IDs, similarly as [Symfony doesn't ](https://symfony.com/doc/current/contributing/code/bc.html).

Reply:
Thanks for the note, indeed a BC makes more sense in that context.

>BTW we don't guarantee BC promise on service definitions, similarly as Symfony doesn't

I was not aware it did not.